ocs_ci/utility/nodemgmt.py

```diff
@@ -0,0 +1,106 @@
+from ocs_ci.framework import config
+from ocs_ci.utility import powernodes
+
+
+class NodeManagement(object):
```

Comment:
Hi @gitsridhar , I might be missing something but how is this base class different from `class NodesBase`?

Reply:
@shylesh good question, NodesBase has methods to add and attach the node to cluster. But if you want to share or inheric logic to add the node, you cannot do it today. Your implementation of NodesBase cannot extend from another implementation of NodesBase (technically possible, but may not be relevant outside of add node - like stop/start nodes etc.). I believe this PR has a clean framework specific to adding and deleting nodes dynamically. NodesBase has no facility for deleting a node and also add and delete additional storage. I expected this comment.

Reply:
@gitsridhar , I still didn't understand completely. If you look at AWS node management in the platfrom_nodes.py: 
``` python
class NodeBase
|____class AWSNodes
         |____class AWSUPINode
```

Here based on the platform tests can instantiate AWSUPINode OR AWSIPINode and perform all the operations like add node, remove node, add_storage, remove_storage etc. This way most common things stay  in `NodeBase` , then `AWSNode` will have something specific to AWS but common to both IPI & UPI , then `AWSUPINode` will have things specific to AWS UPI.

In the same way we can have class for IBMPlatform as well , 
```python
class NodeBase
|____class IBMPlatformNode
        |____class IBMUPINode
```
I am not sure if `IBMUPI` makes sense but just an example, tests can instantiate `IBMUPI()` or `IBMIPI()` and perform operations.

am I missing something completely ?